shell/platform/darwin/ios/framework/Source/FlutterPluginAppLifeCycleDelegate.mm

```diff
@@ -248,6 +248,18 @@ - (void)application:(UIApplication*)application
   }
 }
 
+- (void)application:(UIApplication*)application
+    didFailToRegisterForRemoteNotificationsWithError:(NSError*)error {
+  for (NSObject<FlutterApplicationLifeCycleDelegate>* delegate in _delegates) {
```

Comment:
why not `for (id<FlutterApplicationLifeCycleDelegate> delegate in _delegates) {` ?

and make `FlutterApplicationLifeCycleDelegate` sub-protocol of `NSObject` like:
```objc
@protocol FlutterApplicationLifeCycleDelegate<NSObject>
//...
@end
```
if it's autogenerated by Flutter, I assume it's a sub-protocol of `NSObject`

Reply:
`NSObject` is more specific than `id`, so if anything it would be safer. I guess it doesn't matter too much though.